server/src/features/users/user.controller.js

```diff
@@ -128,4 +130,195 @@ export default class UserController {
+      );
+
+      return res.status(200).json({
+        success: true,
+        message: result.message,
+        data: {
+          userId: result.userId,
+          status: result.status,
+        },
+      });
+    } catch (err) {
+      // Pass the error to the error handling middleware
+      next(err);
+    }
+  }
+
+  /**
+   * Add an event to user's favorites
+   * POST /api/users/favorites
+   * Body: { eventId: string }
+   */
+  static async addToFavorites(req, res, next) {
+    try {
+      const { eventId } = req.body;
+      const userId = req.user._id;
+
+      // Validate request
+      const { error } = favoriteEventSchema.validate({ eventId });
+      if (error) {
+        return res.status(400).json({
+          success: false,
+          message: error.details[0].message,
+        });
+      }
+
+      // Check if user is allowed to add to favorites
+      const allowedRoles = ['student', 'staff', 'ta', 'professor'];
+      if (!allowedRoles.includes(req.user.role)) {
+        return res.status(403).json({
+          success: false,
+          message: 'Only students, staff, TAs, and professors can add events to favorites',
+        });
+      }
+
+      // Check if event exists
+      const event = await mongoose.model('Event').findById(eventId);
+      if (!event) {
+        return res.status(404).json({
+          success: false,
+          message: 'Event not found',
+        });
+      }
+
+      // Add to favorites if not already added
+      const user = await User.findById(userId);
+      if (!user.favoriteEvents.includes(eventId)) {
+        user.favoriteEvents.push(eventId);
+        await user.save();
+      }
+
+      return res.status(200).json({
+        success: true,
+        message: 'Event added to favorites',
+        data: {
+          eventId,
+          favoritesCount: user.favoriteEvents.length,
+        },
+      });
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  /**
+   * Remove an event from user's favorites
+   * DELETE /api/users/favorites/:eventId
+   */
+  static async removeFromFavorites(req, res, next) {
+    try {
+      const { eventId } = req.params;
+      const userId = req.user._id;
+
+      // Validate request
+      const { error } = favoriteEventSchema.validate({ eventId });
+      if (error) {
+        return res.status(400).json({
+          success: false,
+          message: error.details[0].message,
+        });
+      }
+
+      // Check if user is allowed to remove from favorites
+      const allowedRoles = ['student', 'staff', 'ta', 'professor'];
+      if (!allowedRoles.includes(req.user.role)) {
+        return res.status(403).json({
+          success: false,
+          message: 'Only students, staff, TAs, and professors can manage favorites',
+        });
+      }
+
+      // Remove from favorites
+      const user = await User.findByIdAndUpdate(
+        userId,
+        { $pull: { favoriteEvents: eventId } },
+        { new: true }
+      );
+
```

Comment:
```suggestion
const oldUser = await User.findById(userId, { favoriteEvents: 1 });

if (!oldUser.favoriteEvents.includes(eventId)) {
  return res.status(400).json({
    success: false,
    message: 'Event was not in favorites'
  });
}
```
$pull silently does nothing if the value didn't exist.
notify the user that it wasn't in the favorites

Reply:
hwa bytshal mn el favorites in case el heart not filled (not in the favorites) , fa lw 3amal click 3ala el heart w hwa not dimmed automatically hayb2a added wel 3aks  fahma asdy